src/notation/view/noteinputcursor.cpp

```diff
@@ -34,12 +34,21 @@ void NoteInputCursor::paint(mu::draw::Painter* painter)
     NoteInputState state = noteInput->state();
     RectF cursorRect = noteInput->cursorRect();
 
+    constexpr auto widthMultiplier = 1.3;
+    constexpr int leftLineWidth = 3;
+
+    // calculate where the box is, and how to center it with new width
+    auto oldWidth = cursorRect.width();   // store old width
+    auto newWidth = cursorRect.width() * widthMultiplier; // set new width
+    auto pixToSlideLeft = (((oldWidth - newWidth) / 2) - (leftLineWidth / 2)); // calculate new center pos
```

Comment:
All cursor position calculations must be in the NotationNoteInput::cursorRect method, NoteInputCursor only paint.

As for the problem, it seems to me that it is enough for us to do the following in NotationNoteInput::cursorRect 
```
    const mu::engraving::StaffType* staffType = staff->staffType(inputState.tick());
    bool isTabStaff = staffType->isTabStaff();

    int sideMargin = isTabStaff ? 16 : 4;
```

The cursor may look not so elegant, but I would not like to complicate the code, as it will be difficult to maintain.
(But if the designers or testers do not like the new solution, then we will think about your solution)

Thanks

Reply:
not a problem. I'm here to learn. I can see how yours makes more sense. plus yours fixes it at the source and mine was more of a bandaid.